network/p2p/p2ptest/client.go

```diff
@@ -28,63 +29,100 @@ func NewClient(
 	clientNodeID ids.NodeID,
 	serverNodeID ids.NodeID,
 ) *p2p.Client {
-	clientSender := &enginetest.Sender{}
-	serverSender := &enginetest.Sender{}
+	return NewClientWithPeers(
+		t,
+		ctx,
+		clientNodeID,
+		map[ids.NodeID]p2p.Handler{serverNodeID: handler},
+	)
+}
 
+// NewClientWithPeers generates a client to communicate to a set of peers
+func NewClientWithPeers(
```

Comment:
I can make this a separate PR if requested - I've gotten feedback in the past about PRs with test utilities where it might be hard to understand why it's needed without corresponding usage.